### PR TITLE
refactor(genetic): Improved roulette efficiency

### PR DESCRIPTION
**Pull request by javiertury**
Use choose_weighted instead of an infinite loop for increased performance and constant time.

### CODE REVIEW

File: libs/genetic-algorithm/src/selection/roulette_wheel.rs [hunk 1]
```diff
@@ -36,16 +36,9 @@ impl SelectionPolicy for RouletteWheelSelection {
...
-            let individual = population
-                .iter()
-                .choose(rng)
-                .expect("got an empty population");
-
-            if rng.gen_bool((individual.fitness() / self.max_fitness) as _) {
-                return individual;
-            }
-        }
+        return population
```
Comment (Patryk27) [inline libs/genetic-algorithm/src/selection/roulette_wheel.rs]: Would you mind dropping this `return`? Usually it's more idiomatic not to use `return` unless it's necessary.

**Review by Patryk27 — CHANGES_REQUESTED** [review]
Hi! Thank you so much for pointing `choose_weighted()` - I've dropped one comment that should be pretty easy to fix :-)

File: libs/genetic-algorithm/src/selection/roulette_wheel.rs (hunk 1 above)
Comment (javiertury) [reply libs/genetic-algorithm/src/selection/roulette_wheel.rs]: Done!

**Comment by Patryk27** [comment]
Ah, I've refactored some `use`-s in the meantime and we ended up having conflicts - could you please rebase your branch?

**Comment by javiertury** [comment]
Now it's ready

**Comment by javiertury** [comment]
And before this PR is over, let me thank you for creating this project. It covers very interesting topics (deep learning, rust, genetic algorithms) in a clear and simple that anyone can understand.

**Comment by Patryk27** [comment]
Thanks, it means a lot!

**Pull request merged**